doc/ublk.1

```diff
@@ -0,0 +1,273 @@
+'\" t
+.\"     Title: ublk
+.\"    Author: [FIXME: author] [see http://www.docbook.org/tdg5/en/html/author]
+.\" Generator: DocBook XSL Stylesheets vsnapshot <http://docbook.sf.net/>
+.\"      Date: 02/23/2025
+.\"    Manual: ublk: manage ublk devices
+.\"    Source: ublk
+.\"  Language: English
+.\"
+.TH "UBLK" "1" "02/23/2025" "ublk" "ublk: manage ublk devices"
+.\" -----------------------------------------------------------------
+.\" * Define some portability stuff
+.\" -----------------------------------------------------------------
+.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.\" http://bugs.debian.org/507673
+.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
+.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.ie \n(.g .ds Aq \(aq
+.el       .ds Aq '
+.\" -----------------------------------------------------------------
+.\" * set default formatting
+.\" -----------------------------------------------------------------
+.\" disable hyphenation
+.nh
+.\" disable justification (adjust text to left margin only)
+.ad l
+.\" -----------------------------------------------------------------
+.\" * MAIN CONTENT STARTS HERE *
+.\" -----------------------------------------------------------------
+.SH "NAME"
+ublk \- Utility to manage ublk devices
+.SH "SYNOPSIS"
+.HP \w'\fBublk\ <command>\ [<command\ options>]\ [<type\ specific\ options>]\fR\ 'u
+\fBublk <command> [<command options>] [<type specific options>]\fR
+.SH "DESCRIPTION"
+.PP
+ublk is a utility that allows you to create, recover, view or delete user\-space block devices\&.
+.PP
+ublk by default comes with three different types of devices, null, loop and nbd\&.
+.PP
+The following commands are supported:
+.SH "ADD COMMAND"
+.PP
+Command to add a ublk device\&.
+.PP
+\fB add {\-t, \-\-type} TYPE [{\-n, \-\-number} DEV_ID] [{\-q, \-\-queues} NR_HW_QUEUES] [{\-d, \-\-depth} QUEUE_DEPTH] [{\-z, \-\-zero_copy}] [{\-u, \-\-uring_comp} URING_COMP] [{\-g, \-\-need\-get\-data} NEED_GET_DATA] [{\-r, \-\-user_recovery} USER_RECOVERY] [{\-i, \-\-user_recovery_reissue} USER_RECOVERY_REISSUE] [{\-e, \-\-user_recovery_fail_io} USER_RECOVERY_FAIL_IO] [\-\-debug_mask=0x{DBG_MASK}] [\-\-unprivileged] [\-\-usercopy] [\-\-eventfd] [<type specific options>] \fR
+.PP
+\fB\-t, \-\-type\fR
+.RS 4
+Specifies the type of device to create\&. The three types of supported devices are null, loop and bnd\&.
+.RE
+.PP
+\fB\-n, \-\-number\fR
+.RS 4
+Create a device with this id\&. The device node will be /dev/ublkb_n_\&.
+.RE
+.PP
+\fB\-q, \-\-queue\fR
+.RS 4
+Number of queues to create\&. Each queue is services by a dedicated child process\&. Default is 1\&.
+.RE
+.PP
+\fB\-d, \-\-depth\fR
+.RS 4
+Maximum queue\-depthfor each queue\&. Default is 4096\&.
+.RE
+.PP
+\fB\-z, \-\-zero_copy\fR
+.RS 4
+zero copy requires 4k block size, and can remap ublk driver\*(Aqs io request into ublksrv\*(Aqs vm space
```

Comment:
Please drop --zero-copy argument, and this feature is still in-progress, and not merged in kernel side.  I just pushed PR#100 for removing the argument in code.